js/src/rive_advanced.mjs.d.ts

```diff
@@ -168,68 +372,140 @@ interface RiveOptions {
     get workEnd(): number;
     get loopValue(): number;
     get speed(): number;
-    /** Time of the animation in seconds */
+    /**
+     * Number of seconds the animation has advanced by
+     */
     time: number;
+    /**
+     * Flag to determine if the animation looped (this is reset when the loop restarts)
+     */
     didLoop: boolean;
+    /**
+     * Advances/scrubs the LinearAnimationInstance by the set amount of seconds. Note that this only moves the "time" in the animation, but does not apply changes to the properties in the Artboard.
+     * This must be called before the `apply()` method of LinearAnimationInstance.
+     * 
+     * @param sec - Scrub the animation instance by a number of seconds
+     */
     advance(sec: number): any;
     /**
-     * Apply animation on the artboard
+     * Apply a mixing value on the animation instance. This is useful if you are looking to blend multiple animations together and want to dictate a strength for each of the animations played back.
+     * This also applies new values to properties of objects on the Artboard according to the keys of the animation.
+     * This must be called after the `advance()` method of `LinearAnimationInstance`
+     *  
      * @param mix 0-1 the strength of the animation in the animations mix.
      */
     apply(mix: number): any;
-    // Deletes the backing Wasm animation instance
+    /**
+     * Deletes the underlying instance created via the WASM. It's important to clean up this instance when no longer in use
+     */
     delete(): void;
   }
-  
+
+  /**
+   * Rive class representing a StateMachine instance. Use this class to advance and control a particular state machine in the render loop (i.e scrub, grab state machine inputs, set up listener events, etc.).
+   *
+   * Important: Make sure to delete this instance when it's no longer in use via the `delete()` method. This deletes the underlying c++ reference and
+   * frees up the backing WASM object. This can be done in cases where the user navigates away from the page with this animation, the canvas is unmounted, etc.
+   */
   export declare class StateMachineInstance {
+    /**
+     * Create a new StateMachineInstance reference
+     * @param stateMachine - A StateMachineInstance retrieved via the Artboard (i.e `artboard.stateMachineByName('foo')`)
+     * @param artboard - The Artboard instance for this state machine
+     */
     constructor(stateMachine: StateMachineInstance, artboard: Artboard);
     get name(): string;
+    /**
+     * Returns the number of inputs associated with this state machine
+     * @returns Number of inputs
+     */
     inputCount(): number;
-    input(i: number): SMIInput; 
+    /**
+     * Returns the SMIInput at the given index
```

Comment:
I would call it a "State machine input" in the comment, since the word SMIInput already exists in the code, and doesn't necessarily make it clear what it refers to if you're new to Rive

Reply:
ah right, good call!